pkg/collect/cluster_resources.go

```diff
@@ -137,6 +137,14 @@ func ClusterResources(c *Collector) (map[string][]byte, error) {
 		return nil, err
 	}
 
+	// crs
```

Comment:
The `crds` and `crs` functions both call `client.CustomResourceDefinitions().List(ctx, metav1.ListOptions{})`. We could move this call into this function so that it only needs to be called once, and then pass its return value into these two functions.

The disadvantage to this is that it links these two functions that are otherwise independent. Does anyone consider this to be a problem?

Reply:
I understand the problem here. But IMO keeping the func `crs` independent from `crds` will keep the behaviour of the `crds` function intact as it is currently in the upstream troubleshoot package which might in turn help when we raise the upstream PR to provide this functionality in main package. Your thoughts @mhrabovcin and @jongiddy ?

Reply:
Yes, I think that's reasonable for now.